src/include/h5cpp/path.hpp

```diff
@@ -88,32 +90,11 @@ class DLL_EXPORT Path
     //!
     const_iterator end() const;
 
-    //!
-    //! \brief iterator to the first element
-    //!
-    iterator begin();
-
-    //!
-    //! \brief iterator to the last+1 element
-    //!
-    iterator end();
-
-    reverse_iterator rbegin();
-    reverse_iterator rend();
     const_reverse_iterator rbegin() const;
     const_reverse_iterator rend() const;
 
-    void push_front(const value_type &link_name);
-    value_type pop_front();
-
-    void push_back(const value_type &link_name);
```

Comment:
I do not really see the point why we have to remove this. We have to sanatize the input anyhow also with the arithmetic operators. By having `push_back` and `push_front` available we could use STL algorithms to modify the path object. Do I get here something wrong?